### PR TITLE
feat: add ISO support

### PR DESCRIPTION
**Pull request by andrewrynhard**
This reverts commit 3515f4e0f8c11352539ed0d430e1f44f73c8229f.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -454,12 +455,16 @@ FROM alpine:3.11 AS installer
 RUN apk add --no-cache --update \
     bash \
     ca-certificates \
-    cdrkit \
     efibootmgr \
+    grub \
```
Comment (andrewrynhard) [inline Dockerfile:459]: I need to figure out how to compile a font that is required by `grub-mkrescue`. Then we can remove this.

File: cmd/installer/cmd/iso.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+	"github.com/spf13/cobra"
+
+	"github.com/talos-systems/talos/cmd/installer/pkg"
+)
+
+var cfg = []byte(`set default=0
+set timeout=0
+
+menuentry "Talos" {
```
Comment (andrewrynhard) [inline cmd/installer/cmd/iso.go:24]: Maybe "Talos Interactive Installer" is better?

**Review by Ulexus — APPROVED** [review]

File: internal/app/machined/internal/install/install.go [hunk 3]
```diff
@@ -69,7 +69,8 @@ func RunInstallerContainer(disk, platform, ref string, reg config.Registries, op
 	// platform name, this should be determined in the installer container.
 	var config *string
 	if config = procfs.ProcCmdline().Get(constants.KernelParamConfig).First(); config == nil {
-		return fmt.Errorf("no config option was found")
+		c := "none"
+		config = &c
```
Comment (andrewrynhard) [inline internal/app/machined/internal/install/install.go:73]: We no longer require the `talos.config` kernel arg.

**Review by Unix4ever — DISMISSED** [review]
:cool:

**Review by rsmitty — DISMISSED** [review]

File: Dockerfile [hunk 4]
```diff
@@ -447,16 +447,22 @@ WORKDIR /src/cmd/installer
...
     efibootmgr \
+    mtools \
     qemu-img \
     util-linux \
-    xfsprogs
+    xfsprogs \
+    xorriso \
+    xz
 COPY --from=pkg-grub / /
+COPY --from=unicode-pf2 /usr/share/grub/unicode.pf2 /usr/share/grub/unicode.pf2
```
Comment (andrewrynhard) [inline Dockerfile:465]: Temporary workaround until we can get `unicode.pf2` built.

**Pull request merged**